src/utils.ts

```diff
@@ -588,13 +588,18 @@ export async function* syncToLatestBlock(): AsyncGenerator<StatusEvent> {
       }
     }
 
-    // We will send the block only after all transactions were sent
-    // to be sure that all downloads were succesfull since there is no
-    // ROLLBACK yet on the wallet-service.
-    const sendBlockResponse: ApiResponse = await sendTx(preparedBlock);
-
-    if (!sendBlockResponse.success) {
-      logger.debug(sendBlockResponse);
+    try {
+      // We will send the block only after all transactions were sent
+      // to be sure that all downloads were succesfull since there is no
+      // ROLLBACK yet on the wallet-service.
+      const sendBlockResponse: ApiResponse = await sendTx(preparedBlock);
+
+      if (!sendBlockResponse.success) {
+        logger.debug(sendBlockResponse);
+        throw new Error('Erroed sending block to the wallet-service.');
+      }
+    } catch(e) {
+      logger.error(e);
```

Comment:
I see no purpose to throw an error to be captured in its own error handling because the error constructor will use memory for the stack trace without necessity. It would be nice to work with a return object:

```
const sendTxHandler: TxResult = async (preparedBlock) => {
  try {
    await response = await sendTx(preparedBlock);
    if (!response.success) {
      logger.debug(response);
    } 
    return { success: response.success };
  }
  catch(e) {
    logger.error(e);
    return { success: false }
  }
}

const { success } = sendTxHandler(preparedBlock);
if (!success) {
  addAlert...
}
```

Being the exception handler a function `sendTxHandler`, it turns the code more readable and more efficient in relating to the memory.

Reply:
I like the idea! Thanks

Refactor made in 6fce84723454b38da4ddd052ed3ed9ad88778029